### ISSUE TITLE
Request hangs for node-style Readable streams when ContentLength present as an Upload option

### ISSUE THREAD

**Issue by ferbs** [aws/aws-sdk-js-v3#4767]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

I'm streaming an http request body to S3 through lib-storage's Upload class. The proxied upload fails when the actual size exceeds the partSize (5mb by default) AND a Content-Length header is included as an upload option. When this happens, the upload silently hangs until socket timeout or abort signal. (Ironically, it doesn't occur when original request uses Transfer-Encoding chunked since that won't have a content length.)

The problem is easy to work around once understood but impact is severe and it's not easy to identify. 

### SDK version number

@aws-sdk/lib-storage@3.331.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

Node v18

### Reproduction Steps

Set Body to a Readable and include a ContentLength value for an upload larger than 5mb / partSize. Eg:

```javascript
const uploader = new Upload({
    client,
    params: {
      Bucket: 'some-bucket',
      Key: 'some-key,
      ContentType: 'some-mime',
      ContentLength: koaCtx.request.length,
      Body: koaCtx.req,
    },
  });
```

### Observed Behavior

Request hangs until socket timeout 

When logging the request headers in sdk middleware it looks like the full/total object size is applied to each part's Content-Length request header

### Expected Behavior

Upload success

### Possible Solution

Adding a validation check would be a decent enough solution, as would removing ContentLength as an Upload option, or best yet, correctly setting the request header for each part.

**Comment by yenfryherrerafeliz** [comment]
Hi @ferbs, sorry for the late response. I agree that it hangs when using the Upload class and providing the ContentLength parameter. Also the following comment is correct:
> When logging the request headers in sdk middleware it looks like the full/total object size is applied to each part's Content-Length request header

This behavior comes from nodejs http lib which basically hangs when the content-length header value is greater than the length of the provided body.

For now, would it be possible for you to not provide the ContentLength parameter?. However, I will mark this issue as needs-review so it can be evaluated further.

Thanks!

**Comment by ferbs** [comment]
@yenfryherrerafeliz thanks for taking a look. 
While I now know the workaround, it'll still be a horror for the next person. Luckily there are some easy options: 1) throw an error if ContentLength is passed to Upload; or 2) have Upload silently delete the value before making the actual request.

**Comment by f8k8** [comment]
> @yenfryherrerafeliz thanks for taking a look. While I now know the workaround, it'll still be a horror for the next person. Luckily there are some easy options: 1) throw an error if ContentLength is passed to Upload; or 2) have Upload silently delete the value before making the actual request.

This was a horror for us 🙂 In testing, the uploads worked fine as we only used small ones and so didn't get split into multiple parts. It's when they are split that the incorrect ContentLength is passed to the PartUploadCommand. I've opened a PR with a fix.

**Comment by kuhe** [comment]
fix is expected to be released later today as [v3.598.0](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.598.0)
